dev/com.ibm.ws.kernel.service/src/com/ibm/wsspi/kernel/service/utils/AtomicServiceReference.java

```diff
@@ -64,6 +66,8 @@ public class AtomicServiceReference<T> {
 
     private final AtomicReference<ComponentContext> contextRef = new AtomicReference<>();
 
+    private final ReadWriteLock queueLock = new ReentrantReadWriteLock();
```

Comment:
```suggestion
    private final ReadWriteLock referencesLock = new ReentrantReadWriteLock();
```
`queueLock` is probably copy paste issue.   `referencesLock` makes more sense for this class since that is what it is protecting.

Reply:
Well I used queueLock because the thing being protected of type queue ...
` private final PriorityQueue<ReferenceTuple<T>> references = new PriorityQueue<>(2);`

But the name you suggested is more useful, so I have switched to that. 
Also moved the declaration down under the declaration of the protected queue, which seems like a better spot.

Done.